src/Raven.Client/Util/JavascriptConversionExtensions.cs

```diff
@@ -1008,6 +1008,221 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
+                    {
+                        var objReplace = (MemberExpression)mce.Object;
+                        var replaceExprassion = Expression.Parameter(objReplace.Expression.Type, _alias);
+                        var objReplace1 = objReplace.Update(replaceExprassion);
+                        var curmce = mce.Update(objReplace1, mce.Arguments);
+                        writer.Write("includes.document(");
+                        context.Visitor.Visit(curmce);
+                        writer.Write(")");
+                        return;
+                    }
+
+                    if (arg is UnaryExpression { Operand: LambdaExpression arg2 })
+                    {
+                        var targetExpression = (MemberExpression)arg2.Body;
+                        var replaceExprassion = Expression.Parameter(targetExpression.Expression.Type, _alias);
+                        targetExpression = targetExpression.Update(replaceExprassion);
+
+                        writer.Write("includes.document(");
+                        context.Visitor.Visit(targetExpression);
+                        writer.Write(")");
+                        return;
+                    }
+                    throw new NotSupportedException("You can't use the include that way try: let _ = RavenQuery.Include<T>(Expression<Func<T, string>> path)");
+                }
+            }
+        }
+
+        public class IncludeTimeSeriesSupport : JavascriptConversionExtension
+        {
+            public static readonly IncludeTimeSeriesSupport Instance = new IncludeTimeSeriesSupport();
+
+            public override void ConvertToJavascript(JavascriptConversionContext context)
+            {
+                var methodCallExpression = context.Node as MethodCallExpression;
+
+                if (methodCallExpression?.Method.Name != nameof(RavenQuery.IncludeTimeSeries))
+                    return;
+
+                if (methodCallExpression.Method.DeclaringType != typeof(RavenQuery))
+                    return;
+                if (methodCallExpression.Arguments.Count != 2)
+                    return;
+
+                context.PreventDefault();
+
+                var writer = context.GetWriter();
+                using (writer.Operation(methodCallExpression))
+                {
+                    int argNum = methodCallExpression.Arguments.Count;
+                    writer.Write("includes.timeseries(");
+                    for (var i = 0; i < argNum; i++)
+                    {
+                        context.Visitor.Visit(methodCallExpression.Arguments[i]);
+                        if (i < argNum - 1)
+                            writer.Write(",");
+                    }
+                    writer.Write(")");
+                }
+            }
+        }
+
+        public class IncludeCounterSupport : JavascriptConversionExtension
```

Comment:
the code here is almost the same as other Counter JavascriptConversionExtension can we create a single JavascriptConversionExtension for all counter variants?

Reply:
done